### ISSUE TITLE
Error when running "netAnalysis_signalingRole_heatmap"

### ISSUE THREAD

**Issue by HouyuZhang** [sqjin/CellChat#573]
Dear developer,
Thank you for this powerful software to study cell-cell communication. I recently have an error when running the `netAnalysis_signalingRole_heatmap` function, it raised **"Error: elements in col should be named vectors."**
I have checked the `Anno` column and found these cell annotations are factors. 
Does someone have a similar issue when running this function? Thank you in advance.

**Comment by ManuelMoradiellos** [comment]
I do have the same error when running that function and `netVisual_heatmap` too even though I'm using a vector with the same names as my Idents...
At the end I changed the color manually with Illustrator when creating the final version of the plot as I don't know if this error is a bug coming from `ComplexHeatmap`

**Comment by HouyuZhang** [comment]
Yes, I finally found this arose from `ComplexHeatmap`, where the `col` parameter needs a named color vector. 
The format should look like this:
CellType1  CellType2 CellType3
"red" "blue" "black"

**Comment by ManuelMoradiellos** [comment]
Literally I've just noticed that this error arises when you select the `remove.isolate = T`! 

With that option set to FALSE the colors are matched accordingly, and I recommend using it because in the past I saw that it was removing some rows that it shouldn't...

**Comment by ManuelMoradiellos** [comment]
They are the same! In other functions the named vector worked perfectly but in `netVisual_heatmap` it only worked when `remove.isolate = TRUE`; maybe it's because of what you say, as removing the isolates is removing cell types that it cannot find in the named color vector

**Comment by sqjin** [comment]
@ManuelMoradiellos It should be fixed now.
